compiler/rustc_data_structures/src/sync/freeze.rs

```diff
@@ -74,10 +86,10 @@ impl<T> FreezeLock<T> {
 
     #[inline]
     pub fn try_write(&self) -> Option<FreezeWriteGuard<'_, T>> {
```

Comment:
doesn't this function reintroduce the deadlock issue?

Reply:
A bit worse, it allowed mutation after freezing. I changed it back to an earlier variant.

Reply:
Ah, I mistakenly thought we also leaked the lock in addition to setting the frozen atomic